### ISSUE TITLE
EcalRawToDigiGPU/ should work with multiple GPUs

### ISSUE THREAD

**Issue by fwyzard** [cms-patatrack/cmssw#452]
Running with `EcalRawToDigiGPU` on a machine with 2 GPUs results in
```
11-May-2020 23:01:13 CEST  Initiating request to open file file:step2.root
11-May-2020 23:01:14 CEST  Successfully opened file file:step2.root
Begin processing the 1st record. Run 1, Event 8, LumiSection 1 on stream 0 at 11-May-2020 23:01:21.312 CEST
----- Begin Fatal Exception 11-May-2020 23:01:21 CEST-----------------------
An exception of category 'StdException' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 8 stream: 0
   [1] Prefetching for module EcalBarrelRecHitsValidation/'ecalBarrelRecHitsValidation'
   [2] Prefetching for module EcalUncalibRecHitConvertGPU2CPUFormat/'ecalMultiFitUncalibRecHit'
   [3] Prefetching for module EcalCPUUncalibRecHitProducer/'ecalMultiFitUncalibRecHitSoA'
   [4] Prefetching for module EcalUncalibRecHitProducerGPU/'ecalMultiFitUncalibRecHitGPU'
   [5] Calling method for module EcalRawToDigiGPU/'ecalDigisGPU'
Exception Message:
A std::exception was thrown.

/data/user/fwyzard/patatrack/CMSSW_11_1_0_pre7/src/EventFilter/EcalRawToDigi/src/UnpackGPU.cu, line 306:
cudaCheck(cudaMemsetAsync(scratchGPU.pChannelsCounter, 0, sizeof(uint32_t) * 2, cudaStream));
cudaErrorInvalidValue: invalid argument
----- End Fatal Exception -------------------------------------------------
Another exception was caught while trying to clean up files after the primary fatal exception.
11-May-2020 23:01:21 CEST  Closed file file:step2.root

=============================================

MessageLogger Summary

 type     category        sev    module        subroutine        count    total
 ---- -------------------- -- ---------------- ----------------  -----    -----
    1 Additional Exception -s PostGlobalEndRun                       1        1
    2 Fatal Exception      -s PostModuleEvent                        1        1
    3 fileAction           -s file_close                             1        1
    4 fileAction           -s file_open                              2        2

 type    category    Examples: run/evt        run/evt          run/evt
 ---- -------------------- ---------------- ---------------- ----------------
    1 Additional Exceptions PostGlobalEndRun                  
    2 Fatal Exception      1/8                               
    3 fileAction           EndJob                            
    4 fileAction           pre-events       pre-events       

Severity    # Occurrences   Total Occurrences
--------    -------------   -----------------
System                  5                   5
```

**Comment by fwyzard** [comment]
@vkhristenko I think you fixed the HCAL code for this - could you have a look for the ECAL code as well ?

**Comment by vkhristenko** [comment]
addressed with #502

**Comment by fwyzard** [comment]
Thanks :-)
